docs/reference/enrichement.mdx

```diff
@@ -0,0 +1,652 @@
+
+```bash
+$ kubectl gadget run trace_exec -o jsonpretty
+{
+  "args": "/bin/gadgettracermanager -liveness",
+  "comm": "gadgettracerman",
+  "cwd": "",
+  "error": "",
+  "error_raw": 0,
+  "gid": 0,
+  "group": "root",
+  "k8s": {
+    "containerName": "gadget",
+    "hostnetwork": false,
+    "namespace": "gadget",
+    "node": "aks-agentpool-19875870-vmss000001",
+    "owner": {
+      "kind": "DaemonSet",
+      "name": "gadget"
+    },
+    "podName": "gadget-9sjkj"
+  },
+  "loginuid": 4294967295,
+  "loginuser": "uid:4294967295",
+  "mntns_id": 4026532271,
+  "pcomm": "containerd-shim",
+  "pid": 483712,
+  "ppid": 483689,
+  "pupper_layer": false,
+  "runtime": {
+    "containerId": "491cc81598d021253edf154d0bf5d6b7d681f7810d326bde6b9c210506ba793f",
+    "containerImageDigest": "sha256:bcd2a52069e7816acba82c0fdf9b849f8f99bbfe37744a78a3e20240c02d65c1",
+    "containerImageName": "ghcr.io/blanquicet/gadget:latest",
+    "containerName": "gadget",
+    "runtimeName": "containerd"
+  },
+  "sessionid": 4294967295,
+  "tid": 483712,
+  "timestamp": "2024-08-30T12:12:25.621903770Z",
+  "timestamp_raw": 1725019945621903770,
+  "uid": 0,
+  "upper_layer": false,
+  "user": "root"
+}
+```
+
+</TabItem>
+<TabItem value="ig" label="ig">
+
+```bash
+$ ig run trace_exec -o jsonpretty
+{
+  "args": "/bin/gadgettracermanager -liveness",
+  "comm": "gadgettracerman",
+  "cwd": "",
+  "error": "",
+  "error_raw": 0,
+  "gid": 0,
+  "group": "root",
+  "k8s": {
+    "containerName": "gadget",
+    "hostnetwork": false,
+    "namespace": "gadget",
+    "node": "",
+    "podName": "gadget-9sjkj"
+    "owner": {
+      "kind": "",
+      "name": ""
+    },
+  },
+  "loginuid": 4294967295,
+  "loginuser": "uid:4294967295",
+  "mntns_id": 4026532271,
+  "pcomm": "containerd-shim",
+  "pid": 478424,
+  "ppid": 478404,
+  "pupper_layer": false,
+  "runtime": {
+    "containerId": "491cc81598d021253edf154d0bf5d6b7d681f7810d326bde6b9c210506ba793f",
+    "containerImageDigest": "sha256:bcd2a52069e7816acba82c0fdf9b849f8f99bbfe37744a78a3e20240c02d65c1",
+    "containerImageName": "ghcr.io/blanquicet/gadget:latest",
+    "containerName": "gadget",
+    "runtimeName": "containerd"
+  },
+  "sessionid": 4294967295,
+  "tid": 478424,
+  "timestamp": "2024-08-30T12:09:40.622882188Z",
+  "timestamp_raw": 1725019780622882188,
+  "uid": 0,
+  "upper_layer": false,
+  "user": "root"
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Communication with the Kubernetes API server
+
+To communicate with the Kubernetes API server, Inspektor Gadget needs to know
+the configuration to do so.
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+
+`kubectl-gadget` uses the [kubernetes
+cli-runtime](https://github.com/kubernetes/cli-runtime) helpers. This adds
+support for many CLI options that are common to many Kubernetes tools, which let
+us specify how to connect to the cluster, which kubeconfig to use, and so on:
```

Comment:
Even if this is true, it's not related to the enrichment process. Events are enriched by the "gadgettracermanager" process running on each gadget pod, and it talks to the API server by using the credentials that are mounted on that pod.